### PR TITLE
chore: update Quarkus to 2.16.1.Final

### PR DESCRIPTION
**Pull request by vkrizan**
EVNT-812

### CODE REVIEW

File: splunk-quarkus/pom.xml [hunk 1]
```diff
@@ -135,7 +135,6 @@
         <dependency>
             <groupId>io.micrometer</groupId>
             <artifactId>micrometer-registry-prometheus</artifactId>
-            <version>1.8.3</version>
         </dependency>
```
Comment (gwenneg) [inline splunk-quarkus/pom.xml:138]: Did you try replacing this dependency with this?
```
<dependency>
    <groupId>io.quarkus</groupId>
    <artifactId>quarkus-micrometer-registry-prometheus</artifactId>
</dependency>
```

Comment (vkrizan) [reply splunk-quarkus/pom.xml:138]: No I haven't. Is this the correct one?

**Comment by gwenneg** [comment]
FYI There's a Quarkus wiki with migration guides for all Quarkus versions: https://github.com/quarkusio/quarkus/wiki/Migration-Guide-2.16. Considering the huge gap between versions in this PR, it may be a good idea to check everything between `2.8` and `2.16`.

**Comment by vkrizan** [comment]
@gwenneg besides Micrometer returning metrics in Prometheus format, there does not seems to be a breaking change.

**Comment by vkrizan** [comment]
Checking logs of the service with test runs yields:
```
2023-02-06 15:31:28,128 ERROR [com.red.con.not.toC.splunk] (Camel (eventing-splunk) thread #3 - seda://push) IOFailure for event 5f987c17-4279-4783-b420-61cc50e1c802 (orgId REDACTED account REDACTED) to https://REDACTED:8088: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
2023-02-06 15:32:37,016 ERROR [com.red.con.not.toC.splunk] (Camel (eventing-splunk) thread #3 - seda://push) IOFailure for event a16ea4d4-b4d8-469d-bc7a-3822a2d2a5fc (orgId REDACTED account REDACTED) to https://REDACTED:8088: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
```

**Comment by vkrizan** [comment]
@gwenneg PTAL

Tests are failing only on unstable policies test.

**Comment by msager27** [comment]
Unstable policies test or product issue?  The test triggers a policy notification and waits up to 5 mins for it to appear in the event log before checking Splunk.  If it's timing out waiting for the event, the next step is to check the policy logs to see if the notification was sent.

**Comment by vkrizan** [comment]
@msager27 looks like a timeout. The logs on policies backend only have 
```
SRMSG18256: Initialize record store for topic-partition 'platform.inventory.events-43' at position -1.
```
and
```
2023-02-08 09:54:40,037 INFO  [access_log] (executor-thread-1) "REDACTED - eventing-test-1 08/Feb/2023:09:54:40 +0000 "POST /api/policies/v1.0/policies?alsoStore=True HTTP/1.1" 201 299 orgId=REDACTED"
2023-02-08 09:54:40,038 INFO  [access_log] (executor-thread-1) "REDACTED - eventing-test-1 08/Feb/2023:09:54:40 +0000 "POST /api/policies/v1.0/policies?alsoStore=True HTTP/1.1" 201 299 orgId=REDACTED"
2023-02-08 09:59:58,242 INFO  [access_log] (executor-thread-1) "REDACTED - eventing-test-1 08/Feb/2023:09:59:58 +0000 "DELETE /api/policies/v1.0/policies/0382d3c9-ef60-439b-b8f1-af18fd5cf8c7 HTTP/1.1" 200 299 orgId=REDACTED"
2023-02-08 09:59:58,242 INFO  [access_log] (executor-thread-1) "REDACTED - eventing-test-1 08/Feb/2023:09:59:58 +0000 "DELETE /api/policies/v1.0/policies/0382d3c9-ef60-439b-b8f1-af18fd5cf8c7 HTTP/1.1" 200 299 orgId=REDACTED"
``` 
(from the UI container).

**Comment by msager27** [comment]
I reserved an EE namespace and ran the smoke test twice.  All tests passed in both runs.  I looked thru the policy pod logs, but didn't see any notification events logged.  Maybe I'm not looking in the right place.  I also looked at the notification logs.  Here's a sample excerpt:

2023-02-08 21:47:47,473 INFO  [io.sma.rea.mes.kafka] (vert.x-eventloop-thread-1) SRMSG18256: Initialize record store for topic-partition 'platform.notifications.ingress-0' at position -1.
2023-02-08 21:47:47,475 INFO  [com.red.clo.not.eve.EventConsumer] (executor-thread-0) Processing received action [id=null, rh-message-id=a4ee4080-76c0-47da-9fc6-4d7d9eb73bba, orgId=3340851, baet=rhel/policies/policy-triggered]
2023-02-08 21:47:47,484 INFO  [com.red.clo.not.pro.cam.CamelTypeProcessor] (executor-thread-0) Sending CloudEvent [orgId=3340851, integration=iqe-eventing-splunk-3.89.28.35-26d11077-4bbf-400e-a253-b754c06481ad, historyId=ba4c61d8-2533-4e05-b387-bd78cd228f77, originalEventId=a4ee4080-76c0-47da-9fc6-4d7d9eb73bba]
2023-02-08 21:47:47,539 INFO  [com.red.clo.not.eve.FromCamelHistoryFiller] (executor-thread-0) Processing return from camel: {"specversion":"1.0","source":"splunk","type":"com.redhat.console.notifications.history","time":"2023-02-08T21:47:47.535Z","id":"ba4c61d8-2533-4e05-b387-bd78cd228f77","content-type":"application\/json","data":"{\"duration\":650494,\"finishTime\":1675892867535,\"details\":{\"type\":\"com.redhat.console.notification.toCamel.splunk\",\"target\":\"https:\/\/3.89.28.35:8088\"},\"outcome\":\"Success: Event ba4c61d8-2533-4e05-b387-bd78cd228f77 sent successfully\",\"successful\":true}"}

Ideally, it would be good to know if/where Policies logs events since that's more relevant in your case.  My namespace is ephemeral-drfjok and is reserved for 2 days.  Feel free to poke around the logs.

If your EE stays active after a pr_check run, we could perhaps do some troubleshooting live tomorrow.

**Comment by vkrizan** [comment]
@msager27 Thank you Mike! I think we can proceed with this.

Unfortunately Policies do not log much, and the logs within Notifications have overflown. I think that we should improve the Policies logging. Ibutsu (from prod) reveals some error around policy naming in some cases: `Policy name is not unique`.

**Review by gwenneg — APPROVED** [review]
Based on what @msager27 wrote about successful tests runs, LGTM

**Pull request merged**